plugins/modules/ec2_instance.py

```diff
@@ -1630,6 +1631,45 @@ def value_wrapper(v):
     return changes_to_apply
 
 
+def change_instance_metadata_options(instance, params):
+    metadata_options_to_apply = params.get("metadata_options")
+
+    if metadata_options_to_apply is None:
+        return False
+
+    existing_metadata_options = camel_dict_to_snake_dict(instance.get("MetadataOptions"))
+
+    changes_to_apply = {
+        key: metadata_options_to_apply[key]
+        for key in set(existing_metadata_options) & set(metadata_options_to_apply)
+        if existing_metadata_options[key] != metadata_options_to_apply[key]
+    }
+
+    if not changes_to_apply:
+        return False
+
+    request_args = {
```

Comment:
Rather than defaulting to `""` or `None`, I think the values here should default to the current values from `existing_metadata_options`.

Reply:
With the "None" entries being filtered out that *should* result in AWS only altering those parameters which change, we don't need to explicitly pass those options...

Reply:
Yeah I agreed that should be how it works. The boto3/AWS documentation is explicit about that for some of the options and unclear on others, which makes me wary.

Reply:
I am fine to go ahead with either approach.
1. use existing values for metadata options that are not provided in the task
or 
2. assume that boto3 as always will not change any options that are not passed in API call

which one should we go with? I think historically other modules have been using (2)

Reply:
I usually go for 2, but I have no objections to 1, especially if it fixes the integration tests :)

Reply:
going with (1) in this case, with unclear documentation, hopefully tests should pass
ignoring devel sanity